lndclient/router_client.go

```diff
@@ -140,6 +145,23 @@ func (r *routerClient) SendPayment(ctx context.Context,
 	if request.LastHopPubkey != nil {
 		rpcReq.LastHopPubkey = request.LastHopPubkey[:]
 	}
+	if request.KeySend {
+		if request.PaymentHash != nil {
+			return nil, nil, fmt.Errorf(
+				"keysend payment must not include a preset payment hash")
+		}
+
+		var preimage lntypes.Preimage
+		if _, err := rand.Read(preimage[:]); err != nil {
+			return nil, nil, err
+		}
+
+		// Override the payment hash.
+		rpcReq.DestCustomRecords = make(map[uint64][]byte)
+		rpcReq.DestCustomRecords[record.KeySendType] = preimage[:]
```

Comment:
Alternative. Fewer characters but more lines :) Non-blocking ofc
```
rpcReq.DestCustomRecords = map[uint64][]byte {
   record.KeySendType: preimage[:],
}
```

Reply:
done, probably compiles to the same code, but more readable :)